instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-cxf-3.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/CxfHttpServerTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jaxrs.v2_0;
+
+import io.opentelemetry.instrumentation.jaxrs.v2_0.JaxRsHttpServerTest;
+import io.opentelemetry.instrumentation.jaxrs.v2_0.test.JaxRsTestApplication;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import java.util.ArrayList;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import org.apache.cxf.endpoint.Server;
+import org.apache.cxf.jaxrs.JAXRSServerFactoryBean;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class CxfHttpServerTest extends JaxRsHttpServerTest<Server> {
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
+
+  @Override
+  protected Server setupServer() {
+    JAXRSServerFactoryBean serverFactory = new JAXRSServerFactoryBean();
+    JaxRsTestApplication application = new JaxRsTestApplication();
+    serverFactory.setApplication(application);
+    serverFactory.setResourceClasses(new ArrayList<>(application.getClasses()));
+    serverFactory.setProvider(
+        (ExceptionMapper<Exception>)
+            exception ->
+                Response.status(500)
+                    .type(MediaType.TEXT_PLAIN_TYPE)
+                    .entity(exception.getMessage())
+                    .build());
+
+    serverFactory.setAddress(buildAddress().toString());
+
+    Server server = serverFactory.create();
+    server.start();
+
+    return server;
+  }
+
+  @Override
+  protected void stopServer(Server httpServer) {
+    httpServer.stop();
+  }
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    super.configure(options);
+
+    options.setResponseCodeOnNonStandardHttpMethod(500);
```

Comment:
the groovy test [had a conditional to use 500 for testLatestDeps](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/f143d4763d8269b22370f34f5a9541a673e884ad/instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-cxf-3.2/javaagent/src/test/groovy/CxfHttpServerTest.groovy#L46) and 405 otherwise. I see you implemented that in the Jetty test, but that was not needed here as well? I assume not since the tests are all passing

Reply:
It is not needed because of using a lambda in
```java
    serverFactory.setProvider(
        (ExceptionMapper<Exception>)
            exception ->
                Response.status(500)
                    .type(MediaType.TEXT_PLAIN_TYPE)
                    .entity(exception.getMessage())
                    .build());
```
When choosing an exception mapper cxf inspects the exception mapper classes and ends up choosing between a built in mapper and our mapper. When using regular class it picks the built in one and when using lambda our one for the non standard http method test.

Reply:
thank you for the explanation